File: atcoder/abc/python/abc218/abc218_D.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    N = gi()
    X = []; Y = []
    for _ in range(N) : x,y = gis(); X.append(x); Y.append(y)
    pts = set()
    for (x,y) in zip(X,Y) : pts.add(x<<30|y)
    ans = 0
    for i in range(N) :
        x1,y1 = X[i],Y[i] 
        for j in range(N) :
            x2,y2 = X[j],Y[j]
            if x2 <= x1 or y2 <= y1 : continue
            if (x1<<30|y2) not in pts : continue
            if (x2<<30|y1) not in pts : continue
            ans += 1
    sys.stdout.write(str(ans)+'\n')

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

File: abc/go/abc218/abc218_D/abc218_D.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)
var wrtr = bufio.NewWriterSize(os.Stdout, 10_000_000)
var rdr = bufio.NewScanner(os.Stdin)
func gs() string  { rdr.Scan(); return rdr.Text() }
func gi() int     { i,e := strconv.Atoi(gs()); if e != nil {panic(e)}; return i }
func ia(m int) []int { return make([]int,m) }
func fill2(m int) ([]int,[]int) { a,b := ia(m),ia(m); for i:=0;i<m;i++ {a[i],b[i] = gi(),gi()}; return a,b }
type Pt struct {x,y int}
func main() {
	//f1, _ := os.Create("cpu.prof"); pprof.StartCPUProfile(f1); defer pprof.StopCPUProfile()
	defer wrtr.Flush()
	infn := ""; if infn == "" && len(os.Args) > 1 {	infn = os.Args[1] }
	if infn != "" {	f, e := os.Open(infn); if e != nil { panic(e) }; rdr = bufio.NewScanner(f) }
	rdr.Split(bufio.ScanWords); rdr.Buffer(make([]byte,1024),1_000_000_000)
	// PROGRAM STARTS HERE
	N := gi(); X,Y := fill2(N)
	pts := make([]Pt,N); for i:=0;i<N;i++ { pts[i] = Pt{X[i],Y[i]} }
	ptmap := make(map[Pt]bool)
	for _,p := range pts { ptmap[p] = true }
	ans := 0
	for _,p := range pts {
		for _,q := range pts {
			if p.x >= q.x || p.y >= q.y { continue }
			if ptmap[Pt{p.x,q.y}] && ptmap[Pt{q.x,p.y}] { ans++ }
		}
	}
	fmt.Println(ans)
}

